pkg/datastore/projectstore.go

```diff
@@ -203,37 +203,70 @@ func (s *projectStore) UpdateProjectRBACConfig(ctx context.Context, id string, r
 	})
 }
 
-// GetProjectRBACRoles returns the custom rbacs roles and built-in rbac roles.
-func (s *projectStore) GetAllProjectRBACRoles(_ context.Context, _ string) ([]*model.ProjectRBACRole, error) {
-	return nil, ErrUnimplemented
+// GetAllProjectRBACRoles returns the custom rbacs roles and built-in rbac roles.
+func (s *projectStore) GetAllProjectRBACRoles(ctx context.Context, id string) ([]*model.ProjectRBACRole, error) {
+	p, err := s.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	return p.GetAllRBACRoles(), nil
 }
 
 // AddProjectRBACRole adds the custom rbac role.
-func (s *projectStore) AddProjectRBACRole(_ context.Context, _, _ string, _ []*model.ProjectRBACPolicy) error {
-	return ErrUnimplemented
+func (s *projectStore) AddProjectRBACRole(ctx context.Context, id, name string, policies []*model.ProjectRBACPolicy) error {
+	return s.update(ctx, id, func(p *model.Project) error {
+		if err := p.AddRBACRole(name, policies); err != nil {
+			return err
+		}
+		return p.Validate()
```

Comment:
do we need this validation? since `s.update()` validates project object after this updater be applied already. (Same with other Update function)

Reply:
Nice catch! So we don't need those validations.

Reply:
Thanks! addressed by https://github.com/pipe-cd/pipecd/pull/3595/commits/7bc17119dcc504bab20f4aeec3f6beb62ea0aa2d.